schema-engine/connectors/sql-schema-connector/src/sql_doc_parser.rs

```diff
@@ -0,0 +1,788 @@
+                        "some documentation",
+                    ),
+                },
+            )
+        "#]];
+
+        expected.assert_debug_eq(&res);
+    }
+
+    #[test]
+    fn parse_param_13() {
+        use expect_test::expect;
+
+        let res = parse_param(Input("@param      {Int}        $1     some    documentation"));
+
+        let expected = expect![[r#"
+            Ok(
+                ParsedParameterDoc {
+                    alias: None,
+                    typ: Some(
+                        Int,
+                    ),
+                    position: Some(
+                        1,
+                    ),
+                    documentation: Some(
+                        "some    documentation",
+                    ),
+                },
+            )
+        "#]];
+
+        expected.assert_debug_eq(&res);
+    }
+
+    #[test]
+    fn parse_param_14() {
+        use expect_test::expect;
+
+        let res = parse_param(Input("@param {Unknown}  $1"));
+
+        let expected = expect![[r#"
+            Err(
+                ConnectorErrorImpl {
+                    user_facing_error: None,
+                    message: Some(
+                        "SQL documentation parsing: invalid type: 'Unknown' (accepted types are: 'Int', 'BigInt', 'Float', 'Boolean', 'String', 'DateTime', 'Json', 'Bytes', 'Decimal') at '{Unknown}  $1'.",
+                    ),
+                    source: None,
+                    context: SpanTrace [],
+                }
+                SQL documentation parsing: invalid type: 'Unknown' (accepted types are: 'Int', 'BigInt', 'Float', 'Boolean', 'String', 'DateTime', 'Json', 'Bytes', 'Decimal') at '{Unknown}  $1'.
+                ,
+            )
+        "#]];
+
+        expected.assert_debug_eq(&res);
+    }
+
+    #[test]
+    fn parse_sql_1() {
+        use expect_test::expect;
+
+        let res = parse_sql_doc("--   @param      {Int}        $1     some    documentation    ");
+
+        let expected = expect![[r#"
+            Ok(
+                ParsedSqlDoc {
+                    parameters: [
+                        ParsedParameterDoc {
+                            alias: None,
+                            typ: Some(
+                                Int,
+                            ),
+                            position: Some(
+                                1,
+                            ),
+                            documentation: Some(
+                                "some    documentation",
+                            ),
+                        },
+                    ],
+                    description: None,
+                },
+            )
+        "#]];
+
+        expected.assert_debug_eq(&res);
+    }
+
+    #[test]
+    fn parse_sql_2() {
+        use expect_test::expect;
+
+        let res = parse_sql_doc(
+            r#"     --     @description  This query returns a user by it's id
+   --   @param   {Int}   $1:userId  valid   user identifier
```

Comment:
how about we accept typescript types instead?

Reply:
Even just supporting `{Type | null}` using Prisma scalar types would be a considerable improvement, imho